fxa-oauth-server/test/metrics/amplitude.js

```diff
@@ -0,0 +1,156 @@
+    assert.strictEqual(amplitudeModule.length, 2);
+  });
+
+  it('throws if log argument is missing', () => {
+    assert.throws(() => amplitudeModule(null, { clientIdToServiceNames: {} }));
+  });
+
+  it('throws if config argument is missing', () => {
+    assert.throws(() => amplitudeModule({}, { clientIdToServiceNames: null }));
+  });
+
+  describe('instantiate', () => {
+    let amplitude, log;
+
+    beforeEach(() => {
+      log = {
+        error: sinon.spy(),
+        info: sinon.spy()
+      };
+
+      amplitude = amplitudeModule(log, {
+        clientIdToServiceNames: {
+          0: 'amo',
+          1: 'pocket'
+        },
+      });
+    });
+
+    it('interface is correct', () => {
+      assert.isFunction(amplitude);
+      assert.lengthOf(amplitude, 2);
+    });
+
+    describe('empty event argument', () => {
+      beforeEach(() => {
+        return amplitude('', {});
+      });
+
+      it('called log.error correctly', () => {
+        assert.strictEqual(log.error.callCount, 1);
+        assert.lengthOf(log.error.args[0], 2);
+        assert.strictEqual(log.error.args[0][0], 'amplitude.badArgument');
+        assert.deepEqual(log.error.args[0][1], {
+          err: 'Bad argument',
+          event: ''
+        });
+      });
+
+      it('did not call log.info', () => {
+        assert.strictEqual(log.info.callCount, 0);
+      });
+    });
+
+    describe('missing data argument', () => {
+      beforeEach(() => {
+        return amplitude('foo');
+      });
+
+      it('called log.error correctly', () => {
+        assert.strictEqual(log.error.callCount, 1);
+        assert.lengthOf(log.error.args[0], 2);
+        assert.strictEqual(log.error.args[0][0], 'amplitude.badArgument');
+        assert.deepEqual(log.error.args[0][1], {
+          err: 'Bad argument',
+          event: 'foo'
+        });
+      });
+
+      it('did not call log.info', () => {
+        assert.strictEqual(log.info.callCount, 0);
+      });
+    });
+
+    describe('token.created', () => {
+      beforeEach(() => {
+        return amplitude('token.created', {
+          service: '0',
+          uid: 'blee',
+        });
+      });
+
+      it('did not call log.error', () => {
+        assert.strictEqual(log.error.callCount, 0);
+      });
+
+      it('called log.info correctly', () => {
+        assert.strictEqual(log.info.callCount, 1);
+
+        const args = log.info.args[0];
+        assert.strictEqual(args.length, 2);
+        assert.strictEqual(args[0], 'amplitudeEvent');
+        assert.strictEqual(args[1].user_id, 'blee');
+        assert.strictEqual(args[1].event_type, 'fxa_activity - access_token_created');
+        assert.deepEqual(args[1].event_properties, {
+          service: 'amo',
+          oauth_client_id: '0'
+        });
+        assert.deepEqual(args[1].user_properties, {
+          '$append': {
+            fxa_services_used: 'amo'
+          }
+        });
+        assert.ok(args[1].time > Date.now() - 1000);
```

Comment:
I find these kinds of assertion really unhelpful when they fail, because the log message just says "expected false to be truthy". It's more helpful to use the specific assertion where you can, in this case [`assert.isAbove`](https://www.chaijs.com/api/assert/#method_isabove).